checks.d/openstack.py

```diff
@@ -427,20 +499,21 @@ def get_server_stats(self):
         for sid in server_ids:
             self.get_stats_for_single_server(sid)
 
-    def get_all_server_ids(self):
-        url = '{0}/servers'.format(self._nova_url)
+    def get_all_server_ids(self, filter_by_host=None):
+        url = '{0}/servers{1}'.format(self._nova_url, "?host=%s" % filter_by_host if filter_by_host else '')
```

Comment:
Actually, following off the above comments about the agent being scoped to the project, I think we should probably hold off on this. The behavior I'm seeing when `all_tenants=1` and `check_all_servers` is enabled, is that, on a multi-tenant host, the check tries to pull diagnostics for servers outside its authorization scope (i.e. in another project). This leads to `404`s. I suspect this problem will be helped when we move toward allowing multiple auth-scope `instances` per agent, but for now i think it's better to vendor this particular change if you need it.